pulsar-client/src/main/java/org/apache/pulsar/client/impl/transaction/TransactionCoordinatorClientImpl.java

```diff
@@ -168,7 +168,7 @@ public TxnID newTransaction(long timeout, TimeUnit unit) throws TransactionCoord
         try {
             return newTransactionAsync(timeout, unit).get();
         } catch (Exception e) {
-            throw TransactionCoordinatorClientException.unwrap(e);
+            throw TransactionCoordinatorClientException.unwrap(e.getCause());
```

Comment:
Better to check CompletionException in `TransactionCoordinatorClientException.unwrap` ?

Reply:
Good idea! I will modify it immediately